packages/riverpod_lint_flutter_test/test/lints/scoped_providers_should_specify_dependencies.dart

```diff
@@ -121,6 +122,21 @@ void definitelyNotAMain() {
   );
 }
 
+void someTestFunction() {
+  testWidgets('override repositoryProvider in test', (tester) async {
+    await tester.pumpWidget(
+      ProviderScope(
```

Comment:
We need a test where `ProviderScope` is _not_ prevent in pumpWidget

And one where `ProviderScope` is prevent within pumpWidget, but is not the very first widget

Reply:
Wil do, slick!

Reply:
Added some more tests covering these cases.